test/interchaintest/interchainstaking_test.go

```diff
@@ -102,32 +116,239 @@ func TestInterchainStaking(t *testing.T) {
+		"--chain-id", quicksilver.Config().ChainID,
+		"--home", quicksilver.HomeDir(),
+		"--node", quicksilver.GetRPCAddress(),
+	}
+	stdout, _, err = quicksilver.Exec(ctx, queryICA, nil)
+	require.NoError(t, err)
+	parts = strings.SplitN(string(stdout), ":", 2)
+	icaAddr = strings.TrimSpace(parts[1])
+	require.NotEmpty(t, icaAddr)
+
+	version := icatypes.NewDefaultMetadataString("connection-0", "connection-0")
+	_, err = quicksilver.FullNodes[0].ExecTx(
+		ctx, quickUser.KeyName(), "interchain-accounts", "controller", "register", "connection-0",
+		"--version", version,
+	)
+	require.NoError(t, err)
+
+	err = testutil.WaitForBlocks(ctx, 5, quicksilver, gaia)
+	require.NoError(t, err)
+
+	stdout, _, err = quicksilver.Validators[0].ExecQuery(ctx, "interchain-accounts", "controller", "interchain-account", quickUserAddr, "connection-0")
+	require.NoError(t, err)
+	require.NotEmpty(t, stdout)
+
+	var icaQuickUser icacontrollertypes.QueryInterchainAccountResponse
+	err = codec.NewLegacyAmino().UnmarshalJSON(stdout, &icaQuickUser)
+	require.NoError(t, err)
+
+	icaQuickUserAddr := icaQuickUser.Address
+
+	// Bank Send for delegation
+	msgSend := &banktypes.MsgSend{
+		FromAddress: gaiaUserAddr,
+		ToAddress:   icaQuickUserAddr,
+		Amount:      sdk.NewCoins(sdk.Coin{Denom: "uatom", Amount: sdkmath.NewInt(10_000_000)}),
+	}
+
+	cdc := config.EncodingConfig.Codec
+	bz, err := icatypes.SerializeCosmosTx(cdc, []proto.Message{msgSend}) //gfddrg
+	require.NoError(t, err)
+
+	packetData := icatypes.InterchainAccountPacketData{
+		Type: icatypes.EXECUTE_TX,
+		Data: bz,
+		Memo: EncodeValidators(
+			t,
+			[]ValidatorDelegation{
+				{
+					Address: gaiaValidators[0].OperatorAddress,
+					Percent: 50,
+				},
+				{
+					Address: gaiaValidators[1].OperatorAddress,
+					Percent: 25,
+				},
+				{
+					Address: gaiaValidators[2].OperatorAddress,
+					Percent: 25,
+				},
+			},
+		),
+	}
+	jsonPacketData, err := codec.NewLegacyAmino().MarshalJSON(packetData)
+	require.NoError(t, err)
+
+	_, err = quicksilver.FullNodes[0].ExecTx(
+		ctx, quickUser.KeyName(), "interchain-accounts", "controller", "send-tx", "connection-0", string(jsonPacketData),
+	)
+	require.NoError(t, err)
+
+	err = testutil.WaitForBlocks(ctx, 50, quicksilver, gaia)
+	require.NoError(t, err)
+
+	stdout, _, err = quicksilver.Validators[0].ExecQuery(ctx, "interchainstaking", "intent", "gaia-2", quickUserAddr)
+	require.NoError(t, err)
+	require.NotEmpty(t, stdout)
+	t.Logf("Intent: %s", string(stdout))
+
+	stdout, _, err = quicksilver.Validators[0].ExecQuery(ctx, "bank", "balances", quickUserAddr)
+	require.NoError(t, err)
+	require.NotEmpty(t, stdout)
+	t.Logf("User quick bank balances: %s", string(stdout))
+
+	stdout, _, err = gaia.Validators[0].ExecQuery(ctx, "bank", "balances", gaiaUserAddr)
+	require.NoError(t, err)
+	require.NotEmpty(t, stdout)
+	t.Logf("User gaia bank balances: %s", string(stdout))
 }
```

Comment:
Should we be asserting here that the user has 10qAtoms in their quicksilver account, and that there are delegations of 5 atoms and 2.5 atoms to the validators in the map at l301?

Reply:
In my view, that is the 'success' criteria for this test, otherwise we're not really asserting that the deposit has succeeded?

Reply:
Got it, I'm working to fix it.